python/pyspark/sql/readwriter.py

```diff
@@ -289,8 +294,14 @@ def func(iterator):
             keyed._bypass_serializer = True
             jrdd = keyed._jrdd.map(self._spark._jvm.BytesToString())
             return self._df(self._jreader.json(jrdd))
+        elif isinstance(path, DataFrame):
+            if len(path.schema) != 1 or not isinstance(path.schema[0].dataType, StringType):
+                raise TypeError("The DataFrame can contain only one String column: %s"
+                                % path.schema.simpleString())
+            jdataset = path._jdf.__getattr__('as')(self._spark._jvm.Encoders.STRING())
+            return self._df(self._jreader.json(jdataset))
         else:
-            raise TypeError("path can be only string, list or RDD")
+            raise TypeError("path can be only string, list, RDD or DataFrame")
```

Comment:
Eh .. @ueshin, do you actually mind if we see how the discussion would go in https://github.com/apache/spark/pull/23534 before getting this in? I was thinking both are quite related.

Reply:
I'd also say this is working as intended. on the scala side it makes sense to support `Dataset[String]` and an `RDD[String]` is the closest analog here in Pyspark. It's easy to select a column out of a DataFrame anyway.